### PR TITLE
Evaluate clippy tags

### LINKED ISSUE #90: Investigate if RefCell/Rc is the right tool in bindings
https://github.com/iotaledger/wallet.rs/pull/1879#pullrequestreview-1343267526

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

This PR removes as many clippy `allow` tags as possible. I used `itertools` to fix the `tuple_array_conversions` lint. While I am not inclined to add a new dependency for a single use-case, I suspect we can use it to clean up more places.

## Related Issues

Closes #90 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: bindings/nodejs/src/lib.rs [hunk 1]
```diff
@@ -3,7 +3,6 @@
 
 // Increased limit required specifically for call_method
 #![recursion_limit = "256"]
```
Comment (thibault-martinez) [inline bindings/nodejs/src/lib.rs]: Did you try if this was still necessary?

File: bindings/nodejs/src/wallet.rs [hunk 2]
```diff
@@ -99,7 +99,7 @@ pub fn create_wallet(mut cx: FunctionContext) -> JsResult<JsBox<WalletMethodHand
 pub fn call_wallet_method(mut cx: FunctionContext) -> JsResult<JsUndefined> {
     let method = cx.argument::<JsString>(0)?;
     let method = method.value(&mut cx);
-    let method_handler = Arc::clone(&&cx.argument::<JsBox<WalletMethodHandlerWrapper>>(1)?.0);
```
Comment (thibault-martinez) [inline bindings/nodejs/src/wallet.rs:102]: Are you using different clippy rules? Or why wasn't this detected earlier ?

Comment (DaughterOfMars) [reply bindings/nodejs/src/wallet.rs:102]: There was an allow for `needless_borrow`

**Review by thibault-martinez — APPROVED** [review]

**Review by qrayven — APPROVED** [review]

File: sdk/src/client/api/block_builder/input_selection/automatic.rs [hunk 3]
```diff
@@ -139,12 +140,10 @@ impl<'a> ClientBlockBuilder<'a> {
             available_input_addresses.extend(internal.iter().map(|bech32_address| bech32_address.inner));
 
             // Have public and internal addresses with the index ascending ordered.
-            #[allow(clippy::tuple_array_conversions)]
             let public_and_internal_addresses = public
                 .iter()
                 .map(|a| (a, false))
-                .zip(internal.iter().map(|a| (a, true)))
-                .flat_map(|(a, b)| [a, b]);
+                .interleave(internal.iter().map(|a| (a, true)));
```
Comment (kwek20) [inline sdk/src/client/api/block_builder/input_selection/automatic.rs:146]: do we really need a whole crate for this one "optimization"?

Comment (thibault-martinez) [reply sdk/src/client/api/block_builder/input_selection/automatic.rs:146]: Tbh we have been considering itertools for a very long time, I'm just that it'll allow way more than that in the future

**Pull request merged**